### PR TITLE
feat: add macro attr `with_extensions`

### PR DESCRIPTION
**Pull request by niklasad1**
Follow-up on #1306 

Let's add on option to use `Extensions` in the API or not because it may be annoying to just mark it as used and would be nice not to break the previous API for existing users.

```rust
#[rpc(client, server)]
pub trait Rpc {
        // legacy
	#[method(name = "foo"])
	async fn async_method(&self, param_a: u8, param_b: String) -> u16>;

        // with extensions
	#[method(name = "with_ext", with_extensions)]
	async fn f(&self) -> bool;
}

impl RpcServer for () {
	async fn async_method(&self, param_a: u8, param_b: String) -> u16 {
            12
        }

	// NOTE: ext is injected here in the API
        async fn f(&self, ext: &Extensions: b: String) -> {
             ext.get::<u32>().is_ok()
        }
}
```

### CODE REVIEW

**Review by jsdw — APPROVED** [review]
Looks great to me; the attribute works well :)

File: proc-macros/src/render_server.rs [hunk 1]
```diff
@@ -200,20 +222,39 @@ impl RpcDescription {
...
+							})
+						})
+					}
 				} else {
-					handle_register_result(quote! {
-						rpc.register_subscription_raw(#rpc_sub_name, #rpc_notif_name, #rpc_unsub_name, |params, mut pending, context, ext| {
-							#parsing
-							let _ = context.as_ref().#rust_method_name(pending, &ext, #params_seq);
-							#sub_err::None
+					if sub.with_extensions {
```
Comment (pkhry) [inline proc-macros/src/render_server.rs:241]: can you collapse the `if` into `else if` following the recommendation from clippy, otherwise LGTM

**Review by pkhry — APPROVED** [review]

**Comment by mudlee** [comment]
Nice work! May I ask when it'll approximately be released? :)

**Comment by niklasad1** [comment]
> Nice work! May I ask when it'll approximately be released? :)

We are "hoping" to get a new release tomorrow

**Pull request merged**